controller/kubernetes.go

```diff
@@ -171,6 +175,136 @@ func (k *K8s) EventsNamespaces(channel chan SyncDataEvent, stop chan struct{}, i
 	go informer.Run(stop)
 }
 
+func (k *K8s) EventsEndpointSlices(channel chan SyncDataEvent, stop chan struct{}, informer cache.SharedIndexInformer) {
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			item, err := k.convertToEndpointsFromEndpointsSlice(obj, ADDED)
+			if errors.Is(err, ErrIgnored) {
+				return
+			}
+			k.Logger.Tracef("%s %s: %s", ENDPOINTS, item.Status, item.Service)
+			channel <- SyncDataEvent{SyncType: ENDPOINTS, Namespace: item.Namespace, Data: item}
+		},
+		DeleteFunc: func(obj interface{}) {
+			item, err := k.convertToEndpointsFromEndpointsSlice(obj, DELETED)
+			if errors.Is(err, ErrIgnored) {
+				return
+			}
+			k.Logger.Tracef("%s %s: %s", ENDPOINTS, item.Status, item.Service)
+			channel <- SyncDataEvent{SyncType: ENDPOINTS, Namespace: item.Namespace, Data: item}
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			item1, err := k.convertToEndpointsFromEndpointsSlice(oldObj, EMPTY)
+			if errors.Is(err, ErrIgnored) {
+				return
+			}
+			item2, _ := k.convertToEndpointsFromEndpointsSlice(newObj, MODIFIED)
+			if item2.Equal(item1) {
+				return
+			}
+			// fix modified state for ones that are deleted,new,same
+			k.Logger.Tracef("%s %s: %s", ENDPOINTS, item2.Status, item2.Service)
+			channel <- SyncDataEvent{SyncType: ENDPOINTS, Namespace: item2.Namespace, Data: item2}
+		},
+	})
+	go informer.Run(stop)
+}
+
+func (k *K8s) convertToEndpointsFromEndpointsSlice(obj interface{}, status store.Status) (*store.Endpoints, error) {
+
+	getServiceName := func(labels map[string]string) string {
+		return labels["kubernetes.io/service-name"]
+	}
+
+	shouldIgnoreObject := func(namespace string, labels map[string]string) bool {
+		serviceName := getServiceName(labels)
+		if namespace == "kube-system" {
+			if serviceName == "kube-controller-manager" ||
+				serviceName == "kube-scheduler" ||
+				serviceName == "kubernetes-dashboard" ||
+				serviceName == "kube-dns" {
+				return true
+
+			}
+		}
+		return false
+	}
+
+	switch data := obj.(type) {
```

Comment:
They look the same but the types are actually different. There is also some pressure for divergence, for example adding the nodename / zone information to the endpoints will have different handling for v1beta and v1 resources.

Reply:
Sorry I thought I have removed this comment cause it is just irrelevant since we need a concrete type in the end, but I see it slept. 
You can ignore it